### DISCUSSION TITLE
MONAI Application, 3D visor and web applications (Django)

### DISCUSSION THREAD

**Discussion by YerePhy** [Project-MONAI/monai-deploy-app-sdk#306] [Q&A]
Hi,

I'm very new to app development, my background is phyisics and work mainly developing DL models, so be patient with me please. I'd like to know several things:

- **Can MONAI Applications communicate "somehow" with django applications?** What I have in mind is to develop a MONAI App then the django app would send requests to the MONAI APP and the MONAI App should send results to Django App or pop-up a visor with the results or something like that (as you can see I'm totally lost). **Note**: I know very very little about web applications so a comprehensive answer is well recived. 
- **Can MONAI Applications show inference results in OHIF visor?** Like MONAI Label but just showing the inference result

Any resource, explanation or whatever is well recived, thanks beforehand.

**Comment by MMelQin** [comment]
Thank for the questions.

The current design of the MONAI application does not run as a service, so the app reads a discrete payload (e.g. a folder of DICOM instance files), processes the payload with the defined operators in the app including saving the results in the output folder, and then exits.

For the example application, e.g. _Spleen Seg_ and _Liver and Tumor Seg_ , the saved results are DICOM Seg instance files which can be loaded and viewed in any viewers that supports DICOM files, e.g. OHIF viewer and MicroDicom.

**Reply by YerePhy** [reply]
Sorry for the lack of clearness in the questions but I am a complete ignorant about this topic. Could you explain me if the application can be dockerized (with my custom docker file and nvidia base image) and, send to the docker container the input folder and then getting the images in an output folder? Could this docker comunicate with other dockers?

**Reply by MMelQin** [reply]
The MONAI Deploy App SDK provides the Commandline Interface, CLI, which supports command to `exec` a MONAI app simply as a Python script, to `package` the app along with the TorchScript model into a MONAI Application Package essentially a Docker image which is based on Nvidia PyTorch base image, and `run` to run/test the Docker image.

So, the CLI `package` command is the one to use to build the Docker image. Please see the [Jupyter notebook](https://github.com/Project-MONAI/monai-deploy-app-sdk/blob/main/notebooks/tutorials/03_segmentation_app.ipynb)

For now, the MAP Docker communicates with the external entities via file I/O, though it is very much possible to have network communication if one needs and implements it within the app.
